index.js

```diff
@@ -163,15 +163,20 @@ async function decorateRequestAndFastifyInstance(fastify, { asyncInitFunction })
   const { config } = fastify
 
   const ajv = new Ajv({ coerceTypes: true, useDefaults: true })
-  fastify.setSchemaCompiler(schema => ajv.compile(schema))
+  fastify.setValidatorCompiler(({ schema }) => ajv.compile(schema))
 
   fastify.decorateRequest(USERID_HEADER_KEY, config[USERID_HEADER_KEY])
   fastify.decorateRequest(USER_PROPERTIES_HEADER_KEY, config[USER_PROPERTIES_HEADER_KEY])
   fastify.decorateRequest(GROUPS_HEADER_KEY, config[GROUPS_HEADER_KEY])
   fastify.decorateRequest(CLIENTTYPE_HEADER_KEY, config[CLIENTTYPE_HEADER_KEY])
   fastify.decorateRequest(BACKOFFICE_HEADER_KEY, config[BACKOFFICE_HEADER_KEY])
   fastify.decorateRequest(MICROSERVICE_GATEWAY_SERVICE_NAME, config[MICROSERVICE_GATEWAY_SERVICE_NAME])
-  fastify.decorateRequest(ADDITIONAL_HEADERS_TO_PROXY, config[ADDITIONAL_HEADERS_TO_PROXY].split(',').filter(header => header))
+  fastify.decorateRequest(ADDITIONAL_HEADERS_TO_PROXY, null)
+
+  // add mutable references later
+  fastify.addHook('onRequest', async(req) => {
+    req.ADDITIONAL_HEADERS_TO_PROXY = config[ADDITIONAL_HEADERS_TO_PROXY].split(',').filter(header => header)
+  })
```

Comment:
Why this is added later? Is not mutable, the config object contains the env variables. In this way, we recalculate this for every incoming request. It should be restored as previously.

Reply:
What about use `Object.freeze`?

Reply:
I made this change to fix the following warning:

    [FSTDEP006] FastifyDeprecation: You are decorating Request/Reply with a reference type. This reference is shared amongst all requests. Use onRequest hook instead. Property: ADDITIONAL_HEADERS_TO_PROXY

Using `Object.freeze()` does not prevent receiving the warning, since an immutable object is still an object. One alternative to current implementation can be defining an anonymous function that returns the generated array. Though it solves the situation, I am not sure whether it is the best manner to deal with it.

E.g.

    fastify.decorateRequest(ADDITIONAL_HEADERS_TO_PROXY, () => config[ADDITIONAL_HEADERS_TO_PROXY].split(',').filter(header => header))

Reply:
Ok, I see. I don't like the function, it changes the actual usage.

What about a getter? It should preserve the actual usage, avoiding to call the hook for every request. I don't like it so much as solution, but viewing the fastify's PR which add the warning it should work.